server/internal/xorg/xorg.c

```diff
@@ -6,6 +6,69 @@ static char *NAME = ":0.0";
 static int REGISTERED = 0;
 static int DIRTY = 0;
 
+struct linked_list
+{
+    unsigned long number;
+    int keycode;
+    struct linked_list *next;
+};
+
+typedef struct linked_list node;
+node *head=NULL, *last=NULL;
+
+void insertAtLast(unsigned long value, int keycode) {
```

Comment:
Maybe it would be more effective to prepend items instead of appending them. If you press something stacked, then you are releasing it in reverse order. Might be really tiny difference in terms of performance.

Reply:
Since only pressed keys are stored, this list should never have ANY impact on performace.

I do not even know what would be the highest values of items stored in the list. Some Keyboards have a 2-key rollover, some 6-key and some rare n-key.

Reply:
> Since only pressed keys are stored, this list should never have ANY impact on performace.

Calling function in go is using the same mutex for keypress and keyrelease, so we cannot press new keys until the old are released. But we are talking in fraction of seconds, maybe.